fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -648,15 +812,37 @@ def provider_ids
       rescue TransporterRequiresApplicationSpecificPasswordError => ex
         handle_two_step_failure(ex)
         return provider_ids
+      ensure
+        if use_api_key
+          FileUtils.rm_rf(api_key[:key_dir]) unless api_key.nil?
+        end
       end
 
-      lines.map { |line| provider_pair(line) }.compact.to_h
+      @is_altool ? parse_altool_provider_info(lines) : parse_itsm_provider_info(lines)
```

Comment:
It seems to be strange that `ITunesTransporter` knows `is_altool`.
These parser methods should be an abstraction.

How about these changes?

- Remove `is_atool` instance variable.
- Implement `privider_ids` to each executors
    - Just call it of each executor here
- Move `parse_itsm_provider_info` to the base class or Executor class. Call them internally.

Reply:
done! https://github.com/fastlane/fastlane/pull/20631/commits/8be9c7ec4703cc3eb28ae9172a07fd558b9577cf